sotodlib/io/hk_utils.py

```diff
@@ -206,4 +210,78 @@ def get_hkaman(start, stop, config):
     data = get_grouped_hkdata(start, stop, config)
     hk_amans = make_hkaman(data)
     return hk_amans
+
+
+def detcosample_hkamans(hkconfig, detconfig, obs_id):
+    """
+    Given a detector observation, output an HK axismanager that cosampled
+    with detector timestreams.
+
+    Parameters:
+        hkconfig (str): The.yaml filename for loading HK fields and aliases.
+        detconfig (str): The .yaml config file matching readout id and detector ids.
+        obs_id (str): The observation id for a detector UFM. Helps determine
+                      the detector start and stop times needed for interpolation.
+
+    Notes
+    -----
+
+    An example detconfig file looks like:
+
+        data_prefix : "/mnt/so1/data/chicago-latrt"
+        hwp_prefix : "/mnt/so1/shared/site-pipeline/data_pkg/chicago-latrt/hwp/"
+        g3tsmurf_db: "/mnt/so1/shared/site-pipeline/data_pkg/chicago-latrt/g3tsmurf.db"
+    """
+    # load detector timestamps
+    SMURF = ls.G3tSmurf.from_configs(detconfig)
+    session = SMURF.Session()
+    obs = session.query(Observations).filter(Observations.obs_id == obs_id).one()
+    det_aman = ls.load_file([f.name for f in obs.files], archive=SMURF)
+
+    grouped_data = get_grouped_hkdata(obs.start, obs.stop, hkconfig)
+
+    amans = []
+    device_names = []
+    for group in grouped_data:
```

Comment:
This looks like you've copied a lot of code from other functions, ( I think `make_hkaman` ). Straight copies of code in different places are hard to maintain because 1 function gets updated and another doesn't. Can you come up with a way to have one function call the other? Or both functions call a helper?

Reply:
I ended up expanding `make_hkaman()` to be able to handle the 2 cases: (1) output an aman for HK data, organizing by cosampled device where appropriate, and (2) given a detector aman, output an HK aman where the HK data is cosampled to detector timestamps.